### PR TITLE
Better bboxes for scraping using text nodes

### PR DESCRIPTION
**Pull request by ankrgyl**
This builds on a suggestion from @jorenl to use `childNodes` which give us direct access to text nodes, instead of trying to discover these in the DOM itself. It also means that HTML like

```
<div>
  Foo <span> Bar</span>
</div>
```

Can be correctly treated as two distinct nodes (Foo and Bar).

### CODE REVIEW

File: src/docquery/find_leaf_nodes.js [hunk 1]
```diff
@@ -64,13 +23,52 @@ function computeViewport() {
...
+function findLeafNodes(node) {
+  var textNodes = [];
+  var walk = document.createTreeWalker(
+    document.body,
+    NodeFilter.SHOW_TEXT,
+    null,
+    false
+  );
+
+  const vp = computeViewport();
```
Comment (jorenl) [inline src/docquery/find_leaf_nodes.js:35]: I don't think you're using `vp` at all

File: src/docquery/find_leaf_nodes.js [hunk 2]
```diff
@@ -64,13 +23,52 @@ function computeViewport() {
...
+  while (walk.nextNode()) {
+    var node = walk.currentNode;
+    var range = document.createRange();
+    range.selectNodeContents(node);
+    var clientRects = range.getClientRects();
+
+    var rects = [];
+    for (var i = 0; i < clientRects.length; i++) {
+      rects.push(clientRects[i]);
+    }
```
Comment (jorenl) [inline src/docquery/find_leaf_nodes.js:46]: Probably can just do `const rects = Array.from(clientRects)`

Comment (ankrgyl) [reply src/docquery/find_leaf_nodes.js:46]: Ah cool

File: src/docquery/find_leaf_nodes.js (hunk 1 above)
Comment (ankrgyl) [reply src/docquery/find_leaf_nodes.js:35]: Whoops that's before I realized reduce doesn't require an initial argument. Thanks

**Review by jorenl — APPROVED** [review]

**Pull request merged**